include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -154,8 +223,54 @@ class PreconditionerML : public PreconditionerBase<double>
     // re-calculate matrix
     pde_operator.calculate_system_matrix(system_matrix);
 
-    // initialize Trilinos' AMG
-    amg.initialize(system_matrix, ml_data);
+    if(ml_operator_type == MLOperatorType::Laplace)
+    {
+      // Default setup; add near null space for Laplace operator.
+      amg.initialize(system_matrix, ml_data);
+    }
+    else if(ml_operator_type == MLOperatorType::Elasticity)
+    {
+      // get Teuchos::ParameterList to provide custom near null space basis
+      unsigned int const     dimension      = pde_operator.get_matrix_free().dimension;
+      Teuchos::ParameterList parameter_list = get_ML_parameter_list(ml_data, dimension);
```

Comment:
Is there a reason why the parameter list is defined here and not below in line 262/263? It doesn't look like there is any dependency with the code below that computes the rigid body modes?

Reply:
there was no reason, I think I rewrote this part and forgot to move it